distributed/tests/test_executor.py

```diff
@@ -3576,3 +3576,25 @@ def test_idempotence(s, a, b):
 
     yield e._shutdown()
     yield f._shutdown()
+
+
+@gen_cluster(executor=True)
+def test_make_serializable(e, s, a, b):
+    from distributed.executor import serialize_for_publish, deserialize_published
+    import pickle
+    f = delayed(lambda: 1)()
+    futures = yield e._scatter([f, f, f])
+    with pytest.raises(AttributeError):
+        pickle.dumps(futures)
+    with serialize_for_publish(futures) as f:
+        d = pickle.dumps(f)
+    assert d
+
+
+@gen_cluster(executor=True)
+def test_publish_simple(e, s, a, b):
+    data = yield e._scatter(range(12))
+    e.publish_dataset(data, 'data')
+    yield gen.sleep(0.2)
+
+    assert s.published_data
```

Comment:
You should be able to make a pair of executors to test this.  Here is an example test:

```python
@gen_cluster()
def test_two_consecutive_executors_share_results(s, a, b):
    from random import randint
    e = Executor((s.ip, s.port), start=False)
    yield e._start()

    x = e.submit(randint, 0, 1000, pure=True)
    xx = yield x._result()

    f = Executor((s.ip, s.port), start=False)
    yield f._start()

    y = f.submit(randint, 0, 1000, pure=True)
    yy = yield y._result()

    assert xx == yy

    yield e._shutdown()
    yield f._shutdown()
```